File: Practica7/src/Location.java
import java.util.Scanner;

public class Location {
    protected String street;
    protected Integer number;
    protected Integer postalCode;
    protected String province;
    protected String Place;


    public Location(String street, Integer number, Integer postalCode, String province, String place) {
        this.street = street;
        this.number = number;
        this.postalCode = postalCode;
        this.province = province;
        Place = place;
    }

    public String getStreet() {
        return street;
    }

    public void setStreet(String street) {
        this.street = street;
    }

    public Integer getNumber() {
        return number;
    }

    public void setNumber(Integer number) {
        this.number = number;
    }

    public Integer getPostalCode() {
        return postalCode;
    }

    public void setPostalCode(Integer postalCode) {
        this.postalCode = postalCode;
    }

    public String getProvince() {
        return province;
    }

    public void setProvince(String province) {
        this.province = province;
    }

    public String getPlace() {
        return Place;
    }

    public void setPlace(String place) {
        Place = place;
    }


    @Override
    public String toString() {
        return "Location{" +
                "street='" + street + '\'' +
                ", number=" + number +
                ", postalCode=" + postalCode +
                ", province='" + province + '\'' +
                ", Place='" + Place + '\'' +
                '}';
    }

    public static Location createLocation(){
        Scanner s = new Scanner(System.in);
        String locationStreet = s.nextLine();
        Integer locationNumber = s.nextInt();
        Integer locationPostalCode = s.nextInt();
        String locationProvince = s.nextLine();
        String locationPlace = s.nextLine();

        return new Location(locationStreet, locationNumber, locationPostalCode, locationProvince, locationPlace);

    }

}

File: Practica8/Logic/Location.py
class Location:
    def __init__(self, street: str, number: int, postalCode: int, place: str, province: str):
        self.__street = street
        self.__number = number
        self.__postalCode = postalCode
        self.__place = place
        self.__province = province

    def get_street(self) -> str:
        return self.__street

    def set_street(self, street: str):
        self.__street = street

    def get_number(self) -> int:
        return self.__number

    def set_number(self, number: int):
        self.__number = number

    def get_postalCode(self) -> int:
        return self.__postalCode

    def set_postalCode(self, postalCode: int):
        self.__postalCode = postalCode

    def get_place(self) -> str:
        return self.__place

    def set_place(self, place: str):
        self.__place = place

    def get_province(self) -> str:
        return self.__province

    def set_province(self, province: str):
        self.__province = province

    def __str__(self):
        return f"Location(street={self.get_street()}, number={self.get_number()}, postalCode={self.get_postalCode()}, place={self.get_place()}, province={self.get_province()})"
